edxval/tests/test_views.py

```diff
@@ -207,6 +207,86 @@ def test_update_one_of_two_encoded_videos(self):
         )
         self.assertEqual(len(videos[0].encoded_videos.all()), 1)
 
+    def test_update_remove_subtitles(self):
+        # Create some subtitles
+        self._create_videos(constants.COMPLETE_SET_STAR)
+
+        # Sanity check that the subtitles
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), 1)
+        self.assertEqual(len(videos[0].subtitles.all()), 1)
+
+        # Update with an empty list of subtitles
+        url = reverse(
+            'video-detail',
+            kwargs={"edx_video_id": constants.COMPLETE_SET_STAR.get("edx_video_id")}
+        )
+        response = self.client.put(
+            url,
+            dict(subtitles=[], encoded_videos=[], **constants.VIDEO_DICT_STAR),
+            format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Expect that subtitles have been removed
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), 1)
+        self.assertEqual(len(videos[0].subtitles.all()), 0)
+
+    def test_update_remove_encoded_videos(self):
+        # Create some encoded videos
+        self._create_videos(constants.COMPLETE_SET_STAR)
+
+        # Sanity check that the encoded videos were created
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), 1)
+        self.assertEqual(len(videos[0].encoded_videos.all()), 1)
+
+        # Update with an empty list of videos
+        url = reverse(
+            'video-detail',
+            kwargs={"edx_video_id": constants.COMPLETE_SET_STAR.get("edx_video_id")}
+        )
+        response = self.client.put(
+            url,
+            dict(encoded_videos=[], **constants.VIDEO_DICT_STAR),
+            format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Expect that encoded videos have been removed
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), 1)
+        self.assertEqual(len(videos[0].encoded_videos.all()), 0)
+
+    def test_update_courses(self):
+        # Create the video with associated course keys
+        self._create_videos(constants.COMPLETE_SET_WITH_COURSE_KEY)
+
+        # Verify that the video was associated with the courses
+        video = Video.objects.get()
```

Comment:
In this case, I know that there is exactly one video because it was created by the test code (rather than by the code under test).